pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -1383,6 +1383,77 @@ func parseIpspec(ipspec *zconfig.Ipspec,
 		config.DhcpRange.Start = start
 		config.DhcpRange.End = end
 	}
+
+	// take some default
+	if config.DhcpRange.Start == nil || config.DhcpRange.End == nil ||
+		config.Gateway == nil {
+		addrCount := types.GetIPAddrCountOnSubnet(config.Subnet)
+		if addrCount == 0 {
+			return nil
+		}
+		base := types.GetIPSubnetBase(config.Subnet.IP, config.Gateway)
+		ipStart := 1
+		if base != 0 {
+			ipStart = base
+		}
+		dhcpRangeStart := ipStart + 1
+		if addrCount >= types.LargeAppCount {
+			// when gateway is less than the half point,
+			// the dhcpRange starts at the half point,
+			// provided the DhcpRange.End is not set
+			if base < addrCount/2 &&
+				config.DhcpRange.End == nil {
+				dhcpRangeStart = addrCount / 2
+			}
+		}
+		// last addressable endpoint, with 0 base, and subnet.IP as start,
+		// it accounts for (2^(iplen - subnetMask) - 2) addresses
+		dhcpRangeEnd := addrCount - 2
+
+		if config.Gateway == nil {
+			config.Gateway = types.AddToIP(config.Subnet.IP, ipStart)
```

Comment:
The logic above for gateway is overly complex since when Gateway is nil it is always defaulted to 
types.AddToIP(config.Subnet.IP, 1)

Thus I think you should have a separate clause for setting a default gateway to that value on line 1387.